File: forest/siege-of-stonehold.lua
-- using "bash", "cleave"
loop
    f = self:findFlag()
    e = self:findNearest(self:findEnemies())
    if f then
        self:pickUpFlag(f)
    elseif e then
        if e.type == "ogre" and self:isReady("bash") then
            self:bash(e)
        elseif self:isReady("cleave") then
            self:cleave(e)
        else
            self:attack(e)
        end
    end
end

File: forest/siege-of-stonehold.clj
(while true
  (def f (.findFlag this))
  (if f
    (.pickUpFlag this f)
    (when true
      (def e (.findNearestEnemy this))
      (if e
        (if (.isReady this "bash")
          (.bash this e)
          (.attack this e))
        (if (.isReady this "power-up")
          (.powerUp this)
          (.wait this 0.2))))))
